src/pages/index.js

```diff
@@ -68,51 +98,138 @@ cardPreviewPopup.close();
-function createCard(data) {
-    const cardElement = new Card({ data, handleImageClick: (imageData) => {
-        cardPreviewPopup.open(imageData);
+function createCard(data, userId) {
+    const cardElement = new Card({ 
+        data,
+        handleImageClick: (imageData) => {
 
-    }}, selectors.cardTemplate);
+            // Open image popup on click
+            cardPreviewPopup.open(imageData);
+        },
+        handleDeleteClick: () => {
+            
+            // Create the delete popup instance
+            const deletePopup = new PopupWithConfirm(selectors.deletePopup, () => {
 
-    return cardElement.getView();
-}
+                // Handle card deletion
+                api.deleteCard(data._id)
+                .then(() => {
+
+                    // Close the confirmation popup
+                    deletePopup.close();
 
-// Create a section of cards
-const cardSection = new Section(
-    {
-        items: initialCards,
-        renderer: (data) => {
+                    // Delete the card on the page
+                    cardElement.handleDeleteCard();
+                })
+            });
 
-            // Create a new card
-            const cardElement = createCard(data);
+            // Open confirmation popup on click
+            deletePopup.open();
 
-            // Display each card
-            cardSection.addItem(cardElement);
+            // Set the event listeners for the confirmation popup
+            deletePopup.setEventListeners();
         },
-    },
-    selectors.cardsList
-);
+        handleLikeClick: () => {
+
+            // If the user has liked the card, remove the like; vice versa
+            if (cardElement.isLiked()) {
+
+                // Remove like from the server if user has already liked the card
+                api.removeLike(data._id)
+                .then((card) => {
+
+                    // Update like count
+                    cardElement.updateLikeCount(card.likes);
+
+                    // Deactivate like icon
+                    cardElement.removeLikeIcon();
+                })
+            } else {
+
+                // Add like to the server if user has not liked the card
+                api.addLike(data._id)
+                .then((card) => {
+
+                    // Update like count
+                    cardElement.updateLikeCount(card.likes);
+
+                    // Activate like icon
+                    cardElement.addLikeIcon();
+                })
+            }
+        }
+    }, 
+    selectors.cardTemplate, 
+    userId);
+
+    // Display the card
+    return cardElement.getView();
+}
 
-// Render the initial list of cards on the page
-cardSection.renderItems(initialCards);
+let cardSection;
+let userId;
+
+// Get the app's information and make sure promises are loaded in the correct sequence
+api.getAppInfo()
+    .then(([cards, userData]) => {
+
+        // Find the user id
+        userId = userData._id;
+        userName.textContent = userData.name;
```

Comment:
You should use UserInfo class instance methods for updating the profile element contents

Reply:
Done! Thank you.